src/torch_fourier_filter/whitening.py

```diff
@@ -106,60 +131,75 @@ def whitening_filter(
     if power_spec:
         power_spectrum = power_spectrum**2
 
-    if len(image_shape) == 2:
-        rot_avg_mth = rotational_average_dft_2d
-    elif len(image_shape) == 3:
-        rot_avg_mth = rotational_average_dft_3d
+    # output shape is for the image
+    # Handle rfft case for output shape
+    if rfft:
+        last_dim = output_shape[-1] // 2 + 1
+        output_shape_fft = (*output_shape[:-1], last_dim)
+    else:
+        output_shape_fft = output_shape
 
-    # Get 1-dimensional radial average
-    power_spectrum_1d, _ = rot_avg_mth(
-        dft=power_spectrum,
-        image_shape=image_shape,
+    # Add batch dimension if necessary
```

Comment:
Like before, I think making arguments for which dimension(s) to calculate the power spectrum over rather than inferring what the batch dimension and half-dimension (for rfft) would increase usability.